src/components/ViewDeck/ViewDeck.js

```diff
@@ -1,94 +1,99 @@
+  //MANA CALCULATIONS
+  useEffect(() => {
+    const includedCards = cardList.filter(
+      (card) => card.deckid === selectedDeck.id
+    );
+    getManaCalculations(includedCards);
+  }, [cardList]);
+
+  const [includedCards, setIncludedCards] = useState([]);
+  const [cmc, setCmc] = useState([]);
+  const [averageCmc, setAverageCmc] = useState(0);
+  const [quantity, setQuantity] = useState(0);
+  const [featuredUri, setFeaturedUri] = useState("");
+  const [hoverCard, setHoverCard] = useState(
+    "https://i.stack.imgur.com/Vkq2a.png"
+  );
+  const [devotion, setDevotion] = useState({
+    White: 0,
+    Blue: 0,
+    Black: 0,
+    Red: 0,
+    Green: 0,
+    Gray: 0,
+  });
+
+  const NavToEditDeck = (deck) => {
+    history.push(`/editdeck/${deck.id}`);
   };
 
-  cardDisplay = (card) => {
-    //console.log('hovering on', card.name, card, card.api_data);
+  const setDisplayCard = (card) => {
     var parsedData = JSON.parse(card.api_data);
-    //    console.log(parsedData.image_uris.normal);
-
     if (parsedData.image_uris === undefined) {
-      this.setState({
-        hoverCard: `https://s3.thingpic.com/images/Kz/uF3amfCnYFLBggjUNr1sPRKi.jpeg`,
-      });
+      setHoverCard(
+        `https://s3.thingpic.com/images/Kz/uF3amfCnYFLBggjUNr1sPRKi.jpeg`
+      );
     } else {
-      this.setState({
-        hoverCard: parsedData.image_uris.normal,
-      });
+      setHoverCard(parsedData.image_uris.normal);
     }
   };
-
-  removeDisplay = (card) => {
-    // console.log('left:', card.name, card, card.api_data);
-    this.setState({
-      hoverCard: "https://i.stack.imgur.com/Vkq2a.png",
-    });
+  const removeDisplayCard = (card) => {
+    setHoverCard("https://i.stack.imgur.com/Vkq2a.png");
   };
 
-  countQty = (cards) => {
-    if (cards) {
-      var qty = 0;
-      cards.forEach((card) => {
-        //for each instance of a card, qtyToAdd = card.quantity
-        var qtyToAdd = card.quantity;
-        //console.log('qtyToAdd', qtyToAdd);
-        let totalqty = qty + qtyToAdd;
-        qty = totalqty;
-        //var totalCards = qty
-        //console.log('totalCards', totalCards);
-      });
-
-      console.log("new total:", qty);
-
-      if (this.state.deckQty !== qty) {
-        this.setState({
-          deckQty: qty,
-        });
-      }
-    }
+  const getManaCalculations = (cards) => {
+    // setIncludedCards(cards);
+    const qtyToSet = cards.reduce((qty, card) => qty + card.quantity, 0);
+    setQuantity(qtyToSet);
+    const parsedCards = parseCards(cards);
+    setIncludedCards(parsedCards);
+    const devotion = getDevotion(parsedCards);
+    setDevotion(devotion);
+    const featuredCard = cards.filter(
+      (featured) => featured.is_featured === true
+    );
+    const parsedFeaturedCard = featuredCard.map((fCard) =>
+      JSON.parse(fCard.api_data)
+    );
+    const featuredUri = parsedFeaturedCard.map(
+      (fUri) => fUri.image_uris.normal
+    );
```

Comment:
You can actually chain all of this together. 
It would be something like this
```
const featuredUri = cards
      .filter(filterstuff)
      .map(parsestuffs)
      .map(getURIstuffs)

Reply:
oh my god this is going to be so clean